When GlobalExitRoot is not on permissionless it stops with a runtime error: invalid memory address or nil pointer dereference

#### System information

zkEVM Node version: `v0.5.2`
Network: `Testnet`

#### Expected behaviour


#### Actual behaviour


#### Steps to reproduce the behaviour


#### Backtrace

````
{"level":"info","ts":1707125191.6309512,"caller":"synchronizer/synchronizer.go:360","msg":"latestSequencedBatchNumber: 54866, latestSyncedBatch: 54872, lastVerifiedBatchNumber: 54824","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"info","ts":1707125191.6310306,"caller":"synchronizer/synchronizer.go:367","msg":"Syncing trusted state (permissionless)","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"info","ts":1707125191.6310499,"caller":"l2_shared/trusted_batches_retrieve.go:79","msg":"syncTrustedState: Getting trusted state info","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"info","ts":1707125191.7147033,"caller":"l2_shared/trusted_batches_retrieve.go:90","msg":"syncTrustedState: latestSyncedBatch:54872 syncTrustedState:54876","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"info","ts":1707125192.2967913,"caller":"l2_shared/processor_trusted_batch_sync.go:208","msg":"syncTrustedState: batch[54872/54876] mode incremental:  Processing trusted batch: mode=incremental desc=batch exists + intermediateStateRoot matchGER: false, matchSR: false, matchL2Data: false, matchWIP: false, ","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"warn","ts":1707125192.3498013,"caller":"l2_sync_etrog/check_sync_status_to_process_batch.go:73","msg":"We are behind this block number. GER=0x02bfee160869ab199b06c509235c04043aaa87daf93c4d06f6ff9c37152d2b0c sync in block 5223767 and we are in block 5219542 CheckL1SyncStatusEnoughToProcessBatch batchNumber:54872 globalExitRoot: 0x02bfee160869ab199b06c509235c04043aaa87daf93c4d06f6ff9c37152d2b0c ","pid":1,"version":"v0.5.1-3-g9bd7ffea"}
{"level":"error","ts":1707125192.3499508,"caller":"l2_sync_etrog/executor_trusted_batch_sync.go:200","msg":"syncTrustedState: batch[54872/54876] mode incremental: error checkIfWeAreSyncedFromL1ToProcessGlobalExitRoot. Error: must sync from L1%!(EXTRA string=\n/src/log/log.go:142 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()\n/src/log/log.go:251 github.com/0xPolygonHermez/zkevm-node/log.Errorf()\n/src/synchronizer/l2_sync/l2_sync_etrog/executor_trusted_batch_sync.go:200 github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_sync_etrog.(*SyncTrustedBatchExecutorForEtrog).IncrementalProcess()\n/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:220 github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ExecuteProcessBatch()\n/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:159 github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ProcessTrustedBatch()\n/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:130 github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).syncTrustedBatchesToFrom()\n/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:96 github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).SyncTrustedState()\n/src/synchronizer/synchronizer.go:630 github.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).syncTrustedState()\n/src/synchronizer/synchronizer.go:368 github.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).Sync()\n/src/cmd/run.go:326 main.runSynchronizer()\n)","pid":1,"version":"v0.5.1-3-g9bd7ffea","stacktrace":"github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_sync_etrog.(*SyncTrustedBatchExecutorForEtrog).IncrementalProcess\n\t/src/synchronizer/l2_sync/l2_sync_etrog/executor_trusted_batch_sync.go:200\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ExecuteProcessBatch\n\t/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:220\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ProcessTrustedBatch\n\t/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:159\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).syncTrustedBatchesToFrom\n\t/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:130\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).SyncTrustedState\n\t/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:96\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).syncTrustedState\n\t/src/synchronizer/synchronizer.go:630\ngithub.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).Sync\n\t/src/synchronizer/synchronizer.go:368\nmain.runSynchronizer\n\t/src/cmd/run.go:326"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13c3b0b]

goroutine 84 [running]:
github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ExecuteProcessBatch(0xc00059c800, {0x1d34878, 0xc00042f900}, 0xc0002e2f20, {0x1d3f680, 0xc00075d428})
	/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:Z +0x48b
github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*ProcessorTrustedBatchSync).ProcessTrustedBatch(0xc000728688?, {0x1d34878, 0xc00042f900}, 0xc00050c000, {{0xc000360210, 0x2, 0x2}}, {0x1d3f680, 0xc00075d428}, {0xc000aaa720, ...})
	/src/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go:159 +0x4b6
github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).syncTrustedBatchesToFrom(0xc00042f950, {0x1d34878, 0xc00042f900}, 0xd658, 0xd65c)
	/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:130 +0x3e7
github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared.(*TrustedBatchesRetrieve).SyncTrustedState(0xc00042f950, {0x1d34878, 0xc00042f900}, 0xd658)
	/src/synchronizer/l2_sync/l2_shared/trusted_batches_retrieve.go:96 +0x214
github.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).syncTrustedState(...)
	/src/synchronizer/synchronizer.go:630
github.com/0xPolygonHermez/zkevm-node/synchronizer.(*ClientSynchronizer).Sync(0xc00038c380)
/src/synchronizer/synchronizer.go:368 +0x1f57
main.runSynchronizer({0x0, 0x0, 0x0, {{0xc00011ab70, 0xa}, {0xc00011ab80, 0x4}, {0xc00059fe00, 0x1, 0x1}}, ...}, ...)
	/src/cmd/run.go:326 +0x5ab
created by main.start in goroutine 1
	/src/cmd/run.go:226 +0x1c89
````

When submitting logs: please submit them as text and not screenshots.

--- a/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go
+++ b/synchronizer/l2_sync/l2_shared/processor_trusted_batch_sync.go
@@ -218,7 +218,7 @@ func (s *ProcessorTrustedBatchSync) ExecuteProcessBatch(ctx context.Context, pro
 		log.Debugf("%s is partially synchronized but we don't have intermediate stateRoot so it needs to be fully reprocessed", processMode.DebugPrefix)
 		processBatchResp, err = s.Steps.ReProcess(ctx, processMode, dbTx)
 	}
-	if processMode.BatchMustBeClosed {
+	if processBatchResp != nil && err == nil && processMode.BatchMustBeClosed {
 		err = checkProcessBatchResultMatchExpected(processMode, processBatchResp.ProcessBatchResponse)
 		if err != nil {
 			log.Error("%s error verifying batch result!  Error: ", processMode.DebugPrefix, err)

--- a/synchronizer/l2_sync/l2_shared/tests/processor_trusted_batch_sync_test.go
+++ b/synchronizer/l2_sync/l2_shared/tests/processor_trusted_batch_sync_test.go
@@ -1,6 +1,8 @@
 package test_l2_shared
 
 import (
+	"context"
+	"errors"
 	"testing"
 
 	"github.com/0xPolygonHermez/zkevm-node/jsonrpc/types"
@@ -9,6 +11,7 @@ import (
 	"github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared"
 	mock_l2_shared "github.com/0xPolygonHermez/zkevm-node/synchronizer/l2_sync/l2_shared/mocks"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
@@ -320,4 +323,18 @@ func TestGetNextStatusUpdateExecutionResult(t *testing.T) {
 	newStatus, err := testData.sut.GetNextStatus(previousStatus, &ProcessResponse, false, "test")
 	require.NoError(t, err)
 	require.Equal(t, common.HexToHash("0x123"), newStatus.LastTrustedBatches[0].StateRoot)
+}
+
+func TestExecuteProcessBatchError(t *testing.T) {
+	testData := newTestDataForProcessorTrustedBatchSync(t)
+
+	data := l2_shared.ProcessData{
+		Mode:              l2_shared.NothingProcessMode,
+		BatchMustBeClosed: true,
+	}
+	returnedError := errors.New("error")
+	testData.mockExecutor.EXPECT().NothingProcess(mock.Anything, mock.Anything, mock.Anything).Return(nil, returnedError)
+
+	_, err := testData.sut.ExecuteProcessBatch(context.Background(), &data, nil)
+	require.ErrorIs(t, returnedError, err)
 }

--- a/synchronizer/l2_sync/l2_shared/tests/processor_trusted_batch_sync_test.go
+++ b/synchronizer/l2_sync/l2_shared/tests/processor_trusted_batch_sync_test.go
@@ -334,7 +334,6 @@ func TestExecuteProcessBatchError(t *testing.T) {
 	}
 	returnedError := errors.New("error")
 	testData.mockExecutor.EXPECT().NothingProcess(mock.Anything, mock.Anything, mock.Anything).Return(nil, returnedError)
-
 	_, err := testData.sut.ExecuteProcessBatch(context.Background(), &data, nil)
 	require.ErrorIs(t, returnedError, err)
 }

